T3T1 - WalletRecovery

obrusvit opened the pull request:
First batch of changes in recovery flow together with mnemonic keyboards.

More changes TBD.

mmilata reviewed (approved):
Did not test but the changes look very straightforward, let's merge them.

obrusvit commented:
> Did not test but the changes look very straightforward, let's merge them.

@mmilata I rebased on `ui-t3t` and did make more changes to the flows and added more strings to TR after your approve. But everything pretty straight forward.

mmilata reviewed (approved):
https://github.com/trezor/trezor-firmware/actions/runs/8911551406/job/24473121726?pr=3754#step:4:41
^ please also fix before merging
